net/Socket.cpp

```diff
@@ -686,10 +686,10 @@ int SocketPoll::poll(int64_t timeoutMaxMicroS)
                          << " of " << _pollSockets.size() << ") from " << _name);
                 _pollSockets[i] = nullptr;
             }
-            else if( _pollSockets[i]->checkRemoval(newNow) )
+            else if( _pollSockets[i]->checkRemoval(newNow) && _pollSockets[i]->isClosed() )
```

Comment:
Very unclear to me why we are adding even more complexity into the center of our already not-entirely obvious & somewhat complex poll loop. Surely that can be done in 'checkRemoval' (?)

Not clear on the 'isClosed'/isOpen semantic we have here - surely these are still open at the time we want to remove them (?)

Reply:
Yes it could - had it this way earlier but then the method API doc got a bit .. err .. lengthily ..
OK will revise.

Reply:
done